client/fingerprint/env_aws.go

```diff
@@ -128,25 +475,52 @@ func (f *EnvAWSFingerprint) Fingerprint(request *FingerprintRequest, response *F
 		response.AddAttribute(key, v)
 	}
 
-	// newNetwork is populated and added to the Nodes resources
-	var newNetwork *structs.NetworkResource
+	// accumulate resource information, then assign to response
+	var resources *structs.Resources
+	var nodeResources *structs.NodeResources
 
 	// copy over network specific information
 	if val, ok := response.Attributes["unique.platform.aws.local-ipv4"]; ok && val != "" {
 		response.AddAttribute("unique.network.ip-address", val)
-
-		newNetwork = &structs.NetworkResource{
-			Device: "eth0",
-			IP:     val,
-			CIDR:   val + "/32",
-			MBits:  f.throughput(request, ec2meta, val),
+		nodeResources = new(structs.NodeResources)
+		nodeResources.Networks = []*structs.NetworkResource{
+			{
+				Device: "eth0",
+				IP:     val,
+				CIDR:   val + "/32",
+				MBits:  f.throughput(request, ec2meta, val),
+			},
 		}
+	}
+
+	// copy over CPU speed information
+	if specs := f.lookupCPU(ec2meta); specs != nil {
+		response.AddAttribute("cpu.modelname", specs.model)
+		f.logger.Debug("lookup ec2 cpu model name", "model", specs.model)
 
-		response.NodeResources = &structs.NodeResources{
-			Networks: []*structs.NetworkResource{newNetwork},
+		response.AddAttribute("cpu.frequency", fmt.Sprintf("%.0f", specs.mhz))
+		f.logger.Debug("lookup ec2 cpu frequency", "MHz", log.Fmt("%.0f", specs.mhz))
+
+		response.AddAttribute("cpu.numcores", fmt.Sprintf("%d", specs.cores))
+		f.logger.Debug("lookup ec2 cpu cores", "cores", specs.cores)
+
+		if ticks := specs.ticks(); request.Config.CpuCompute <= 0 {
+			response.AddAttribute("cpu.totalcompute", fmt.Sprintf("%d", ticks))
+			f.logger.Debug("setting ec2 cpu ticks", "ticks", ticks)
+			resources = new(structs.Resources)
+			resources.CPU = ticks
+			if nodeResources == nil {
+				nodeResources = new(structs.NodeResources)
+			}
+			nodeResources.Cpu = structs.NodeCpuResources{CpuShares: int64(ticks)}
 		}
+	} else {
+		f.logger.Warn("no cpu specification found for this instance type")
```

Comment:
We've been moving to `failed to ...` pattern, so maybe:

```suggestion
		f.logger.Warn("failed to find the cpu specification for this instance type")
```